standard/ranges.md

```diff
@@ -0,0 +1,291 @@
+# 24 Extended Indexing and Slicing
+
+> <span style="color:red">**Review Note:** This new chapter, currently (§24), is placed here temporarily to avoid text changes due to renumbering occurring in chapters & clauses otherwise unaffected by the PR. It’s final placement is not yet determined, however between the  Arrays ([§17](arrays.md#17-arrays)) and Interfaces ([§18](interfaces.md#18-interfaces)) chapters might be suitable – other placements can be suggested during review. It can be relocated later with just a simple edit to `clauses.json`.</span>
+
+## 24.1 General
+
+This chapter introduces a model for *extended indexable* and *sliceable* *collection* types built on:
+
+- The types introduced in this chapter, `System.Index` (§24.2) and `System.Range` (§24.3);
+- The pre-defined unary `^` (§hat-operator) and binary `..` (§range-operator) operators; and
+- The *element_access* expression.
+
+Under the model a type is classified as:
+
+- a *collection* if it represents a group of *element*s
+- an *extended indexable* collection if it supports an *element_access* expression which has a single argument expression of type `Index` which returns and/or sets a single element of the type, either by value or by reference; and
+- an *extended sliceable* if it supports an *element_access* expression which has a single argument expression of type `Range` which returns a *slice* of the elements of the type by value.
+
+> *Note*: The model does not require that a slice, unlike an element, of the type can be set, but a type may support it as an extension of the model. *end note*
+
+The model is supported for single-dimensional arrays (§12.8.12.2) and strings (§string-access).
+
+The model can be supported by any class, struct or interface type which provides appropriate indexers (§15.9) which implement the model semantics.
+
+Implicit support for the model is provided for types which do not directly support it but which provide a certain *pattern* of members (§24.4). This support is pattern-based, rather than semantic-based, as the semantics of the type members upon which it is based are *assumed* – the language does not enforce, or check, the semantics of these type members.
+
+### 24.1.1 Definitions
```

Comment:
It looks odd to have only one subclause, 24.1.1, under 24,1. I suggest we either

1. push the content of 24.1.1 up to the end of 24.1, or
2. make 24.1.1 24.2 instead.
3.